src/libServer/StatusServer.cpp

```diff
@@ -481,4 +485,32 @@ string StatusServer::GetValidateDB() {
   }
 
   return result;
-}
\ No newline at end of file
+}
+
+bool StatusServer::SetVoteInPow(const std::string& proposalId,
+                                const std::string& voteValue,
+                                const std::string& remainingVoteCount,
+                                const std::string& startDSEpoch,
+                                const std::string& endDSEpoch) {
+  if (LOOKUP_NODE_MODE) {
+    throw JsonRpcException(RPC_INVALID_REQUEST, "Not to be queried on lookup");
+  }
+  if (proposalId.empty() || voteValue.empty() || remainingVoteCount.empty() ||
+      startDSEpoch.empty() || endDSEpoch.empty()) {
+    return false;
+  }
+  try {
+    if (!m_mediator.m_node->StoreVoteUntilPow(proposalId, voteValue,
```

Comment:
With current setup, `m_govProposalInfo` can be overwritten on receiving new proposal through api and can potentially result in undefined behavior.

Reply:
For now we can allow miner to change the vote settings anytime. As long as `m_mutexGovProposal` is acquired wherever `m_govProposalInfo` is used, then that should avoid any race condition and undefined state.

Reply:
Yes.. I agree. I should have said unexpected behaviour instead. Anyways if miners are made aware of this restriction on one proposal voting each ds epoch, it should be fine.
Later we can consider adding this restriction at code level or remove this restriction itself to allow multiple proposal voting in same ds epoch.